gemgis/vector.py

```diff
@@ -19,100 +20,522 @@
 from gemgis.raster import sample
 from gemgis.utils import set_extent
+from scipy.interpolate import griddata, Rbf
+from typing import Union, List, Tuple, Optional, Any
 __all__ = [geometry]
 
+try:
+    import rasterio
+except ModuleNotFoundError:
+    raise ModuleNotFoundError('No valid rasterio installation found')
+
 pd.set_option('display.float_format', lambda x: '%.2f' % x)
 
-# v0.2.0 - Refactoring API
-# def extract_xy_linestrings
-# def extract_xy_points
 
-# Function tested
+def extract_xy_linestrings(gdf: gpd.geodataframe.GeoDataFrame,
+                           reset_index: bool = True,
+                           drop_id: bool = True,
+                           drop_index: bool = True,
+                           drop_points: bool = True,
+                           overwrite_xy: bool = False,
+                           target_crs: str = None,
+                           bbox: List[Union[int, float]] = None) -> gpd.geodataframe.GeoDataFrame:
+    """
+   Extracting x,y coordinates from a GeoDataFrame (LineStrings) and returning a GeoDataFrame with x,y
+   coordinates as additional columns
+   Args:
+       gdf (gpd.geodataframe.GeoDataFrame): GeoDataFrame created from vector data containing elements of geom_type
+       LineString
+       reset_index (bool): Variable to reset the index of the resulting GeoDataFrame, default True
+       drop_id (bool): Variable to drop the id column, default True
+       drop_index (bool): Variable to drop the index column, default True
+       drop_points (bool): Variable to drop the points column, default True
+       overwrite_xy (bool): Variable to overwrite existing X and Y values, default False
+       target_crs (str, pyproj.crs.crs.CRS): Name of the CRS provided to reproject coordinates of the GeoDataFrame
+       bbox (list): Values (minx, maxx, miny, maxy) to limit the extent of the data
+   Return:
+       gdf (gpd.geodataframe.GeoDataFrame): GeoDataFrame with appended x,y columns and optional columns
+   """
+
+    # Checking that gdf is of type GepDataFrame
+    if not isinstance(gdf, gpd.geodataframe.GeoDataFrame):
+        raise TypeError('Loaded object is not a GeoDataFrame')
+
+    # Check that all entries of the gdf are of type LineString
+    if not all(gdf.geom_type == 'LineString'):
+        raise TypeError('All GeoDataFrame entries must be of geom_type linestrings')
+
+    # Checking that the bbox is of type None or list
+    if not isinstance(bbox, (type(None), list)):
+        raise TypeError('The bbox values must be provided as list')
+
+    # Checking that the bbox list only has four elements
+    if isinstance(bbox, list) and not len(bbox) == 4:
+        raise ValueError('Provide minx, maxx, miny and maxy values for the bbox')
+
+    # Checking that all elements of the list are of type int or float
+    if isinstance(bbox, list) and not all(isinstance(i, (int, float)) for i in bbox):
+        raise TypeError('Bbox values must be of type float or int')
```

Comment:
```suggestion
    if bbox is not None:
        if not isinstance(bbox, Sequence):
            raise TypeError('The bbox values must be provided as a sequence')

        # Checking that the bbox list only has four elements
        if len(bbox) != 4:
            raise ValueError('Provide minx, maxx, miny and maxy values for the bbox')

        # Checking that all elements of the list are of type int or float
        if not all(isinstance(i, (int, float)) for i in bbox):
            raise TypeError('Bbox values must be of type float or int')
```

Reply:
We could go further and use [pep 3141](https://www.python.org/dev/peps/pep-3141/), but I think that it's better to just `isinstance(X, (int, float))` in this case, as it's not as complex as checking for protocols.